### PR TITLE
ENH new tasks

### PR DESCRIPTION
**Pull request by etyl**
Follow up on #5 but without the FMRI dataset to simplify the things

### CODE REVIEW

File: solvers/u-net.py [hunk 1]
```diff
@@ -19,52 +21,82 @@ class Solver(BaseSolver):
 
     requirements = []
 
-    def set_objective(self, train_dataset, physics):
-        batch_size = 2
+    def set_objective(self, train_dataset, physics, image_size, dataset_name):
+        batch_size = 1
```
Comment (tomMoral) [inline solvers/u-net.py]: Should be a parameter

File: solvers/dpir.py [hunk 2]
```diff
@@ -16,36 +24,68 @@ class Solver(BaseSolver):
...
-        self.model = dinv.optim.DPIR(sigma=best_sigma, device=self.device)
+            self.model = model_class(sigma=best_sigma, device=self.device)
         self.model.eval()
 
     def get_result(self):
         return dict(model=self.model, model_name="DPIR", device=self.device)
+
+
+# Custom DPIR solver with 2 channels
+class DPIR_2C(BaseOptim):
```
Comment (tomMoral) [inline solvers/dpir.py]: Remove all things linked to 2 channels, this is MRI stuff

File: solvers/u-net.py [hunk 3]
```diff
@@ -19,52 +21,82 @@ class Solver(BaseSolver):
...
-        )
+        self.device = get_device()
         self.physics = physics.to(self.device)
+        self.image_size = image_size
+        self.dataset_name = dataset_name
 
     def run(self, n_iter):
         epochs = 4
 
+        x, y = next(iter(self.train_dataloader))
```
Comment (tomMoral) [inline solvers/u-net.py]: All this should go in `set_objective`.
This is the setup, it does not really make sense to have it in the initialization time of the run IMO

File: solvers/u-net.py [hunk 4]
```diff
@@ -19,52 +21,82 @@ class Solver(BaseSolver):
...
+        criterion = dinv.loss.SupLoss(metric=dinv.metric.MSE())
+
+        for epoch in range(epochs):
+            model.train()
+            running_loss = 0.0
+
+            for x, y in self.train_dataloader:
+                x, y = x.to(self.device), y.to(self.device)
+
+                if type(self.physics) is dinv.physics.blur.Downsampling:
```
Comment (tomMoral) [inline solvers/u-net.py]:
```suggestion
                if isinstance(self.physics, dinv.physics.blur.Downsampling):
```

File: solvers/u-net.py [hunk 5]
```diff
@@ -19,52 +21,82 @@ class Solver(BaseSolver):
...
+                    pad_top = diff_h // 2
+                    pad_bottom = diff_h - pad_top
+                    pad_left = diff_w // 2
+                    pad_right = diff_w - pad_left
+
+                    y = F.pad(
+                        y,
+                        pad=(pad_left, pad_right, pad_top, pad_bottom),
+                        value=0
+                    )
```
Comment (tomMoral) [inline solvers/u-net.py]: Why is padding needed?

File: objective.py [hunk 6]
```diff
@@ -52,56 +56,83 @@ def set_data(self,
...
+        self.image_size = image_size
 
     def evaluate_result(self, model, model_name, device):
         # The keyword arguments of this function are the keys of the
         # dictionary returned by `Solver.get_result`. This defines the
         # benchmark's API to pass solvers' result. This is customizable for
         # each benchmark.
 
-        batch_size = 2
+        batch_size = 1
```
Comment (tomMoral) [inline objective.py]: move as parameter

File: objective.py [hunk 7]
```diff
@@ -114,4 +145,8 @@ def get_objective(self):
         # for `Solver.set_objective`. This defines the
         # benchmark's API for passing the objective to the solver.
         # It is customizable for each benchmark.
-        return dict(train_dataset=self.train_dataset, physics=self.physics)
+
+        return dict(train_dataset=self.train_dataset,
+                    physics=self.physics,
+                    image_size=self.image_size,
+                    dataset_name=self.dataset_name,)
```
Comment (tomMoral) [inline objective.py]: if possible avoid passing dataset_name.
Also, image_size should be retrieved through `train_dataset`
